packages/postgres-database/src/simcore_postgres_database/migration/versions/17eea8ba5d4f_new_trashed_colums.py

```diff
@@ -0,0 +1,39 @@
+"""new  trashed colums
+
+Revision ID: 17eea8ba5d4f
+Revises: fce5d231e16d
+Create Date: 2024-11-05 18:07:13.399757+00:00
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "17eea8ba5d4f"
+down_revision = "fce5d231e16d"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column(
+        "folders_v2", sa.Column("trashed_at", sa.DateTime(timezone=True), nullable=True)
+    )
+    op.add_column(
+        "folders_v2",
+        sa.Column(
+            "trashed_explicitly",
```

Comment:
what means trashed_explicitely?
does that mean the trashing process is done and the resource is in the trash? does it mean it was deleted? pretty unclear

Reply:
In the column description (below) it says:

> Indicates whether the folder was explicitly trashed by the user (true) or inherited its trashed status from a parent (false).

Reply:
When a folder is trashed, both the folder and all its contents are marked as `trashed`. At this point, none of these items appear on the dashboard. However, only items that were **explicitly** trashed (like the folder itself) will be listed in the trash bin. Items that are *implicitly* trashed (e.g., files within a trashed folder) will not appear individually in the trash bin.

As a result, only **explicitly** trashed items can be restored from the UI. This approach aligns with the one used in [Google Drive](https://developers.google.com/drive/api/reference/rest/v3/files#File). While we have the backend functionality to support restoring implicitly trashed items (similar to Dropbox), implementing it in the UI requires additional development, so we’ve opted for a simpler solution in this initial release.